pkgs/development/lisp-modules/ql.nix

```diff
@@ -0,0 +1,209 @@
+    });
+    trivial-ssh-libssh2 = super.trivial-ssh-libssh2.overrideLispAttrs (o: {
+      nativeLibs = [ pkgs.libssh2 ];
+    });
+    sqlite = super.sqlite.overrideLispAttrs (o: {
+      nativeLibs = [ pkgs.sqlite ];
+    });
+    cl-libuv = super.cl-libuv.overrideLispAttrs (o: {
+      nativeBuildInputs = [ pkgs.libuv ];
+      nativeLibs = [ pkgs.libuv ];
+    });
+    cl-liballegro = super.cl-liballegro.overrideLispAttrs (o: {
+      nativeLibs = [ pkgs.allegro5 ];
+    });
+    cl-ode = super.cl-ode.overrideLispAttrs (o: {
+      nativeLibs = let
+        ode' = pkgs.ode.overrideAttrs (o: {
+          configureFlags = [
+            "--enable-shared"
+            "--enable-double-precision"
+          ];
+        });
+      in [ ode' ];
+    });
+    classimp = super.classimp.overrideLispAttrs (o: {
+      nativeLibs = [ pkgs.assimp ];
+    });
+    sdl2 = super.sdl2.overrideLispAttrs (o: {
+      nativeLibs = [ pkgs.SDL2 ];
+    });
+    lispbuilder-sdl-cffi = super.lispbuilder-sdl-cffi.overrideLispAttrs (o: {
+      nativeLibs = [ pkgs.SDL ];
+    });
+    cl-opengl = super.cl-opengl.overrideLispAttrs (o: {
+      nativeLibs = [ pkgs.libGL ];
+    });
+    cl-glu = super.cl-glu.overrideLispAttrs (o: {
+      nativeLibs = [ pkgs.libGLU ];
+    });
+    cl-glut = super.cl-glut.overrideLispAttrs (o: {
+      nativeLibs = [ pkgs.freeglut ];
+    });
+    cl-glfw = super.cl-glfw.overrideLispAttrs (o: {
+      nativeLibs = [ pkgs.glfw ];
+    });
+    cl-glfw-opengl-core = super.cl-glfw-opengl-core.overrideLispAttrs (o: {
+      nativeLibs = [ pkgs.libGL ];
+    });
+    cl-glfw3 = super.cl-glfw3.overrideLispAttrs (o: {
+      nativeLibs = [ pkgs.glfw ];
+    });
+    lev = super.lev.overrideLispAttrs (o: {
+      nativeLibs = [ pkgs.libev ];
+    });
+    cl-rdkafka = super.cl-rdkafka.overrideLispAttrs (o: {
+      nativeBuildInputs = [ pkgs.rdkafka ];
+      nativeLibs = [ pkgs.rdkafka ];
+    });
+    cl-async-ssl = super.cl-async-ssl.overrideLispAttrs (o: {
+      nativeLibs = [ pkgs.openssl ];
+    });
+    iolib = super.iolib.overrideLispAttrs (o: {
+      nativeBuildInputs = [ pkgs.libfixposix ];
+      nativeLibs = [ pkgs.libfixposix ];
+      systems = [ "iolib" "iolib/os" "iolib/pathnames" ];
+    });
+    cl-ana_dot_hdf-cffi = super.cl-ana_dot_hdf-cffi.overrideLispAttrs (o: {
+      nativeBuildInputs = [ pkgs.hdf5 ];
+      nativeLibs = [ pkgs.hdf5 ];
+      NIX_LDFLAGS = [ "-lhdf5" ];
+    });
+    gsll = super.gsll.overrideLispAttrs (o: {
+      nativeBuildInputs = [ pkgs.gsl ];
+      nativeLibs = [ pkgs.gsl ];
+    });
+    cl-libyaml = super.cl-libyaml.overrideLispAttrs (o: {
+      nativeLibs = [ pkgs.libyaml ];
+    });
+    cl-libxml2 = super.cl-libxml2.overrideLispAttrs (o: {
+      nativeLibs = [ pkgs.libxml2 ];
+    });
+    cl-readline = super.cl-readline.overrideLispAttrs (o: {
+      nativeLibs = [ pkgs.readline ];
+    });
+    md5 = super.md5.overrideLispAttrs (o: {
+      lispLibs = [ super.flexi-streams ];
```

Comment:
What exactly is happening there that _Lisp_ deps need to be corrected?

Reply:
Fixes build on non-SBCL

`md5` uses `sb-rotate-byte` on SBCL and uses `flexi-streams` on other implementations. I guess Quicklisp is built using SBCL and that's why the dependency is missing.

Reply:
now there is (-: https://github.com/quicklisp/quicklisp-projects/issues/2266

Reply:
Thanks

It's more complicated than you decribe: with ql:quickload it will still work because deps are just hints and QL hooks into ASDF… (which is why _completely_ broken dependencies could persist for mutiple QL releases in a row)